enterprise/utils.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+"""
+Utility functions for enterprise app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+
+def get_available_idps():
+    """
+    Get available Identity Providers.
+
+    Raises a ValueError if it third_party_auth app is not available.
+
+    Return:
+        a list of SAMLProviderConfig instances if third_party_auth app is available.
+    """
+    try:
+        # We give a choice field to user only if SAMLProviderConfig is present, otherwise we show
+        # an integer field, Since, we will be adding a custom enterprise admin so we will also be removing
+        # this dependency.
+        from third_party_auth.models import SAMLProviderConfig
+        return SAMLProviderConfig.objects.current_set().filter(enabled=True).all()
+    except ImportError:
+        raise ValueError("SAMLProviderConfig is not available.")
+
+
+def get_idp_choices():
+    """
+    Get a list of identity providers choices for enterprise customer.
+
+    Return:
+        A list of choices of all identity providers, None if it can not get any available identity provider.
+    """
+    first = [("", "-"*7)]
+    try:
+        return first + [(idp.idp_slug, idp.name) for idp in get_available_idps()]
```

Comment:
@saleem-latif @asadiqbal08 Yes, you should definitely use the `provider_id` and not `idp_slug`. We may want to support other types of providers in the future (OAuth2, LDAP, ...), and it will be much easier to support that in the future if we use the generic ID `provider_id` and not the SAML-specific one.

Likewise, the `get_available_idps()` function is not the best way to approach this. You can just use:
```
try:
    from third_party_auth.provider import Registry
except ImportError:
    Registry = None

def get_idp_choices():
    choices = [("", "-"*7)]
    if Registry:
        choices += [(idp.provider_id, idp.name) for idp in Registry.enabled()]
    return choices
```

If we really need to only show SAML providers in this list, you can use `Registry.get_enabled_by_backend_name("tpa-saml")` instead of `Registry.enabled()`. I don't think there's any harm in also supporting other provider types now, but I could be missing something.

`Registry` is the public API to the third party auth backends and it should be used rather than accessing database tables directly, as in `SAMLProviderConfig.objects.current_set().filter(enabled=True).all()`. The Registry API also ensures that the resulting backends are sorted by name, which makes the admin nicer.



Three other notes:

1. Are you sure it's necessary to use `first = [("", "-"*7)]`? I thought that if `blank=True` on a field, the django forms library would generate the blank entry automatically?
1. When you declare the field `identity_provider = models.SlugField(null=True, blank=True, default=None)` in `models.py`, it should say what the allowed values are (`provider_id` values). Right now it doesn't indicate what sort of slugs should be stored.
1. Did this PR add a check to make sure that the IdP's `site` matches the EC's `site`? That's important and should raise an error if it's different.